hbase-client/src/main/java/org/apache/hadoop/hbase/shaded/protobuf/ProtobufUtil.java

```diff
@@ -1552,13 +1571,23 @@ public static ComparatorProtos.Comparator toComparator(ByteArrayComparable compa
   public static ByteArrayComparable toComparator(ComparatorProtos.Comparator proto)
     throws IOException {
     String type = proto.getName();
-    String funcName = "parseFrom";
     byte[] value = proto.getSerializedComparator().toByteArray();
+
     try {
+      ByteArrayComparable result = COMPARATORS.getAndCallByName(type, value);
+      if (result != null) {
+        return result;
+      }
+
+      if (!ALLOW_FAST_REFLECTION_FALLTHROUGH) {
+        throw new IllegalStateException("Failed to deserialize comparator " + type
+          + " because fast reflection returned null and fallthrough is disabled");
+      }
```

Comment:
Should we add an else clause that ticks up a metric? Or maybe an emitted-once-per-`type` WARN that notifies the operator that we've fallen back to the slower path? I don't yet understand how often this might occur. Same for Filters.

Reply:
I only added this so that I could write tests, since this is all static methods.

I don't think we want a warn or counter here. How often it happens will depend on the usage of custom filters. If they don't use custom filters, this will never fail. If they use exclusively custom filters, then it will fail every time. It's not really a failure mode, more backwards compatibility handling.